### PR TITLE
Upgrade debian-base to 0.3.1 for CVEs

### PR DESCRIPTION
**Pull request by satyasm**
**What this PR does / why we need it**:
Upgrade debian-base to 0.3.1 in response to CVE fixes in debian-base

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Bumps up the version number of related components.

**Release note**:

```release-note
Bump up version number of debian-base, debian-hyperkube-base and debian-iptables. 
Also updates dependencies of users of debian-base. 
debian-base version 0.3.1 is already available.
```

### CODE REVIEW

File: build/debian-hyperkube-base/Makefile [hunk 1]
```diff
@@ -19,11 +19,11 @@
 
 REGISTRY?=staging-k8s.gcr.io
 IMAGE?=debian-hyperkube-base
-TAG=0.10
+TAG=0.11
```
Comment (cblecker) [inline build/debian-hyperkube-base/Makefile]: Has this image been pushed?
```
$ docker pull k8s.gcr.io/debian-hyperkube-base-amd64:0.11
Error response from daemon: manifest for k8s.gcr.io/debian-hyperkube-base:0.11 not found
```

File: build/debian-iptables/Makefile [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 REGISTRY?="staging-k8s.gcr.io"
 IMAGE=debian-iptables
-TAG=v10
+TAG=v11
```
Comment (cblecker) [inline build/debian-iptables/Makefile]: Has this image been pushed?
```
$ docker pull k8s.gcr.io/debian-iptables-amd64:v11
Error response from daemon: manifest for k8s.gcr.io/debian-iptables-amd64:v11 not found
```

**Review by cblecker — COMMENTED** [review]
/lgtm cancel
/hold

**Comment by satyasm** [comment]
/cc @tallclair 

@cblecker not sure what you are expecting here. My understanding of the process is as follows when we want to upgrade debian base.

1. Do a PR for the newer version and once merged build and push the new tag.
2. Once image from 1 is available, do a PR (this one) for the all the dependents of debian-base and once merged, build and push the new tags.
3. Once images from 2 is available, do a PR of the dependent images from 2 and once merged, build and push the new tags.
4. Once images from 3 are available, do the final update of the installation manifests and add-ons to refer to the updated tags. 

In this case, due to some oversight, a PR for 1 was not done but the image was pushed. So I'm doing the back fill of the tag update for 1 as part of this PR. So the upgraded images for debian-iptables and debian-base-hyperkube will be built and pushed once this PR is approved and merged.

**Comment by cblecker** [comment]
I'll defer to @ixdy on the proper order. IIRC, we build/push, then PR. Either way, I'd like to see this documented more.

**Comment by tallclair** [comment]
I don't think the order is well defined. What matters is:

1. Update the FROM base image tag (locally)
2. Build & Push the image
3. Update the manifests to the new tag (& merge)

The parts that don't matter are:

- when the changes from (1) get merged - This is just in order to reflect the current state in the open source. I'd prefer to do this before (2), so that the pushed image is always built from merged code.
- updating the tag - This is purely symbolic, as our image promotion process ends up relabeling the image. The ordering here doesn't matter, so to decrease the number of PRs, I'd prefer to do this in the same PR as the above.

Given this, I think the current PR looks good.

File: build/debian-iptables/Makefile (hunk 2 above)
Comment (ixdy) [reply build/debian-iptables/Makefile]: this one is trickier, since v10.1 makes no sense. v10-r2? bleh.

**Comment by satyasm** [comment]
Thanks @ixdy, btw since I'm not sure, does the hold need to be removed?

File: build/debian-iptables/Makefile (hunk 2 above)
Comment (ixdy) [reply build/debian-iptables/Makefile]: though we did have a v5.1. (also a v5, and a v5.0?)

**Comment by ixdy** [comment]
@satyasm yes, though can we maybe rename those tags to .1 instead of +1?

**Comment by satyasm** [comment]
yup, can do that if that makes more sense. Will update the PR.

**Comment by satyasm** [comment]
@ixdy updated debian-hyperkube-base to 0.10.1 and debian-iptables to v10.1 instead of 0.11 and v11 respectively.

**Comment by neolit123** [comment]
/sig release
(?)

**Pull request merged**